lib/core/src/Cardano/Wallet/DB/MVar.hs

```diff
@@ -125,10 +130,16 @@ newDBLayer = do
                         Right $ Just $ Database cp meta (txs' <> txs) k
             txs' `deepseq` alterMVar db alter key
 
-        , readTxHistory = \key -> let
-                sortedTxHistory = sortOn slot . Map.toList . txHistory
-                slot = Down . slotId . snd . snd
-            in maybe mempty sortedTxHistory . Map.lookup key <$> readMVar db
+        , readTxHistory = \key sortDir filt -> let
+                sort' = case sortDir of
+                    Ascending -> id
+                    Descending -> reverse
```

Comment:
I tried 
```haskell
direction = case sortDir of
    Ascending -> id
    Descending -> Down
```
but it doesn't type-check since `a -> a` and `a -> Down a` are different types.

> So, I guess we don't really care

That's what I thought too, however:

I think `sortBy` and `comparing` should work. I'll try that.

Reply:
Ah, yeay. Maybe `Down . Down` :man_shrugging: . But, doesn't matter much here. We should definitely remove that code...